cmd/vic-machine/converter/converter_test.go

```diff
@@ -0,0 +1,611 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package converter
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/url"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/lib/config"
+	"github.com/vmware/vic/lib/install/data"
+	"github.com/vmware/vic/lib/install/validate"
+	"github.com/vmware/vic/pkg/vsphere/extraconfig"
+)
+
+type testFinder struct {
+}
+
+func (f *testFinder) ObjectReference(ctx context.Context, ref types.ManagedObjectReference) (object.Reference, error) {
+	return testObject{
+		Common: object.NewCommon(nil, ref),
+	}, nil
+}
+
+type testObject struct {
+	object.Common
+}
+
+func (c testObject) ObjectName(ctx context.Context) (string, error) {
+	switch c.Common.Reference().String() {
+	case "DistributedVirtualPortgroup:dvportgroup-357":
+		return "management", nil
+	case "DistributedVirtualPortgroup:dvportgroup-358":
+		return "vm-network", nil
+	case "DistributedVirtualPortgroup:dvportgroup-55":
+		return "bridge", nil
+	case "DistributedVirtualPortgroup:dvportgroup-56":
+		return "external", nil
+	default:
+		return "unknown", nil
+	}
+}
+
+func TestMain(t *testing.T) {
+	//	ConverterLogLevel = log.DebugLevel
```

Comment:
Because the converter log is disabled by default, and not set through general log level, this comments is used to show samples. 😄